modules/farm/farm_livestock/farm_livestock_weight/farm_livestock_weight.module

```diff
@@ -132,6 +132,11 @@ function farm_livestock_weight_dlwg($asset) {
       // Calculate time difference.
       $timediff = $latest_log->timestamp - $previous_log->timestamp;
       $timediff_days = round($timediff/86400,2);
+      
+      // When multiple inputs are entered on a single day, round $timediff_days up to 1.
+      if ($timediff_days == 0) {
+        $timediff_days = 1;
+      }
```

Comment:
Kudos for jumping in and fixing bugs you've found!

Take all this with a grain of salt since I don't personally use the weight tracking in farmOS right now and acknowledge there's a lot to be said for ensuring that the page doesn't crash as a first level of bug fix...

---

Besides the division by zero error, it seems like this logic was a little broken to start with and I suspect this change introduces as many issues as it fixes.

For example, integer division by 86400 isn't a great way to bucket timestamps into days because timestamps which are a few seconds apart can get counted as being different days while timestamps which are 23h59s apart can get counted as being the same.

From a different perspective, honoring time zones to bucket the weight logs into days wouldn't necessarily produce the right results either. With that approach, a log at 11 PM would be counted as being a "day later" than a log at 6 AM even though they're only 7 hours apart.

Similarly, this change would produce pretty odd behavior. Suppose you had a measurement for a flock of animals 3 days ago and had just entered new measurements for them today, but accidentally weighed and logged two measurements for one of the animals. All the other animals would show the correct weight gain, but the one with a duplicate log would presumably just show the variance in the scale's measurements and claim that was the "daily weight gain" for that animal.

---

It's more work, and I don't know what the stakeholders care about with this stat, but I would argue a better strategy would be to have a window heuristic where this logic averages datapoints which are close together and looks for two datapoints (or averages thereof) which are a day or more apart, then uses floating point division to calculate the "daily weight gain" from those datapoints.

I'm curious whether @mstenta can comment on who the stakeholders are and if they can be looped in on what the expected behavior is?